.github/workflows/publish.yml

```diff
@@ -40,17 +38,10 @@ jobs:
       - name: Setup Gradle
         uses: gradle/actions/setup-gradle@v3
         with:
-          cache-read-only: ${{ github.ref != 'refs/heads/main' && github.ref != 'refs/heads/arrow-2' }}
+          cache-read-only: ${{ github.ref != 'refs/heads/main' && github.ref != 'publish-2.0.0-alpha.1' }}
 
       - name: Assemble
-        run: ./gradlew assemble -Pversion=${{ inputs.version }} -Penable_wasm=true
-
-      - name: Upload reports
-        if: failure()
-        uses: actions/upload-artifact@v4
-        with:
-          name: 'reports-${{ matrix.os }}'
-          path: '**/build/reports/**'
+        run: ./gradlew assemble -Pversion=2.0.0-alpha.3 -Penable_wasm=true
 
-      - name: Publish next major SNAPSHOT version
-        run: ./gradlew -Pversion=${{ inputs.version }} publish
+      - name: Publish artifacts
+        run: ./gradlew -Pversion=2.0.0-alpha.3 publish -Penable_wasm=true
```

Comment:
I think `--no-configuration-cache` fixes it (https://vanniktech.github.io/gradle-maven-publish-plugin/central/#automatic-release)